### ISSUE TITLE
Minor anoyance and potenttially major inaccuracy, with the way PRTs identify which inputs they need.

### ISSUE THREAD

**Issue by aharjula** [maths/moodle-qtype_stack#194]
[Here](https://github.com/maths/moodle-qtype_stack/blob/master/stack/potentialresponsetree.class.php#L271-L274) we have a slight problem that limits question authors from using input names with different case (derivatives come to mind) in situations where the different case inputs would go to different trees. It also does not really handle situations where the code contains strings with those sub-strings. 

Annoys me a bit and the solution to that inaccuracy of that variable-name identification would require a bit more serious parser than that reg-ex so this will probably remain unsolved, but yet another example where such an parser might be a nice thing to have. The quick fix would be to just strip the potential strings from the cas-strings before matching and not do that case insensitivity thing at all. But dropping the case insensitivity would probably break some legacy stuff, so I won't push any changes.

Basically, means that the users of the state system that works with strings as identifiers for the variables need to name their variables so that they do not collide with input names, but as they typically have a logical connections to each other this is inconvenient.

**Comment by timhunt** [comment]
It's actually more serious than that. Most Moodle installs using MySQL have a much more serious problem with things that differ only by case https://tracker.moodle.org/browse/MDL-36208 / https://docs.moodle.org/dev/Database_collation_issue.

If STACK is insisting that all variables are distinguishable, even if compared case-insensitively, then that is saving question authors from serious pain, even if it is also occasionally a bit irritating.

**Comment by aharjula** [comment]
A valid point that there are people using databases that are not case sensitive. But the other issue is the strings inside cas-strings, which has no such excuse.

Anyway as our systems run on a case-sensitive databases and our authors have written STACK questions for a decade with case-sensitive variable names I am very much against STACK starting to require that the variables are case-insensitively distinguishable so that is not an option.
